src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensRangeHandler.cs

```diff
@@ -64,5 +74,40 @@ public SemanticTokensRangeHandler(IGlobalOptionService globalOptions)
 
             return new RoslynSemanticTokens { Data = tokensData, IsFinalized = isFinalized };
         }
+
+        private async Task<bool> IsDataFinalizedAsync(Project project, CancellationToken cancellationToken)
+        {
+            // If the project's compilation isn't yet available, kick off a task in the background to
+            // hopefully make it available faster since we'll need it later to compute tokens.
+            if (!_projectIdToCompilation.ContainsKey(project.Id))
+            {
+                var compilationTask = project.GetCompilationAsync(cancellationToken);
```

Comment:
I believe that this task should only be kicked off _after_ isFullyLoaded = true.  

As is, it looks like the code could kick off the compilation task potentially before we've fully loaded - so we may get a compilation, but a compilation created from an inaccurate view of the projects.